### DISCUSSION TITLE
Objective with a discrete time value of a continuous variable.

### DISCUSSION THREAD

**Discussion by IlyaOrson** [infiniteopt/InfiniteOpt.jl#181] [Q&A] [answered]
Is it possible to set one specific time value of a continuous time variable in the objective function? Or maybe a set of these discrete time points? Aiming to cover the special case where the optimization maximizes or minimizes just the final state of the system.

Additionally, could this be combined with the integral objective to cover mixed scenarios?

**Comment by pulsipher** [comment, marked as answer]
Yes, we can accomplish this via the restricted variable syntax for defining point/semi-infinite variables (https://pulsipher.github.io/InfiniteOpt.jl/v0.5/guide/variable/#Restricted-Variables). For instance, we could define an objective that has a terminal variable and an integral over the whole domain as well:
```
using InfiniteOpt
model = InfiniteModel()
@infinite_parameter(model, t in [0, 1], num_supports = 10)
@variable(model, y, Infinite(t))
@objective(model, Min, y(1) + integral(y^2, t))
```

Note this syntax is quite flexible to allow you to embed any point variables that you wish to add, including a sum over particular time points.
